sdks/uniswapx-sdk/integration/test/V3DutchOrder.spec.ts

```diff
@@ -303,17 +311,436 @@ describe.only("DutchV3Order", () => {
+            .deadline(deadline)
+            .swapper(swapper.address)
+            .nonce(NONCE)
+            .startingBaseFee(BigNumber.from(0))
+            .input({
+                token: tokenIn.address,
+                startAmount: SMALL_AMOUNT,
+                curve: {
+                    relativeBlocks: [1],
+                    relativeAmounts: [BigInt(0)],
+                },
+                maxAmount: SMALL_AMOUNT,
+                adjustmentPerGweiBaseFee: BigNumber.from(0),
+            })
+            .output({
+                token: tokenOut.address,
+                startAmount: SMALL_AMOUNT,
+                curve: {
+                    relativeBlocks: [1],
+                    relativeAmounts: [BigInt(0)],
+                },
+                recipient: swapperAddress,
+                minAmount: SMALL_AMOUNT,
+                adjustmentPerGweiBaseFee: BigNumber.from(0),
+            })
+            .buildPartial()
+
+        const { domain, types, values } = order.permitData();
+        const signature = await swapper._signTypedData(domain, types, values);
+        const cosignerData = await getCosignerData({
+            inputOverride: SMALL_AMOUNT.div(2)
+        });
+
+        const cosignerHash = order.cosignatureHash(cosignerData);
+        const cosignature = ethers.utils.joinSignature(
+            cosigner._signingKey().signDigest(cosignerHash)
+        );
+
+        const fullOrder = V3DutchOrderBuilder.fromOrder(order)
+            .cosignerData(cosignerData)
+            .cosignature(cosignature)
+            .build();
+
+        const swapperTokenInBalanceBefore = await tokenIn.balanceOf(swapperAddress);
+        const fillerTokenInBalanceBefore = await tokenIn.balanceOf(fillerAddress);
+        const swapperTokenOutBalanceBefore = await tokenOut.balanceOf(swapperAddress);
+        const fillerTokenOutBalanceBefore = await tokenOut.balanceOf(fillerAddress);
+
+        const res = await reactor
+            .connect(filler)
+            .execute(
+                { 
+                    order: fullOrder.serialize(), 
+                    sig: signature
+                }
+            );
+        const receipt = await res.wait();
+        expect(receipt.status).to.equal(1);
+
+        const amountIn = SMALL_AMOUNT.div(2);
+        expect((await tokenIn.balanceOf(swapperAddress)).toString()).to.equal(
+            swapperTokenInBalanceBefore.sub(amountIn).toString()
+        );
+        expect((await tokenIn.balanceOf(fillerAddress)).toString()).to.equal(
+            fillerTokenInBalanceBefore.add(amountIn).toString()
+        );
+
+        expect((await tokenOut.balanceOf(swapperAddress)).toString()).to.equal(
+            swapperTokenOutBalanceBefore.add(SMALL_AMOUNT)
+        );
+        expect((await tokenOut.balanceOf(fillerAddress)).toString()).to.equal(
+            fillerTokenOutBalanceBefore.sub(SMALL_AMOUNT)
+        );
+    });
+
+    it("executes a serialized order with decay", async () => {
```

Comment:
Could you also add a test for a multi-point decay with different values?

Reply:
added!